src/ComicK/ComicKParser.ts

```diff
@@ -171,6 +162,67 @@ export const parseChapters = (
     }
 }
 
+const filterChaptersByScore = (chapterData: ChapterData[], chapters: ChapterData[]): void => {
```

Comment:
Have you checked the performance hit of this filter? Try running the chapter score filtering on a manga with 3000+ chapters

Reply:
any suggestions on which to try? I tried on one piece which seems to have around 1700 chapters and didn’t see a noticeable performance hit there. Not sure what 3000 chapter manga exist 😅

Reply:
https://comick.ink/comic/00-martial-peak

Reply:
ok it’s definitely slightly slower - maybe a second slower. That said, even with the setting toggled off it took about 5 seconds for me, so 5s vs 6s is not a huge difference

Reply:
Not sure if this data structure will be more efficient but instead of saving ChapterData again, why not save the original index of the chapter in chapterData in your map and loop through those indexes when pushing them back into the Chapter[] array

Reply:
oh, so instead of
```
const chapterMap = new Map<number, { score: number, chapter: ChapterData }>()
```
I should do 
```
const chapterMap = new Map<number, { score: number, chapterIndex: number }>()
```
? I can give that a shot. I'd be very surprised if it's any faster but less memory pressure is probably a good thing 🤷

Reply:
ok updated, though it did require a force unwrap, which scares me a lot - I generally avoid those like the plague when writing swift code, not sure if they're more common in javascript. Performance with the 3000 chapter manga you linked is the same.  


Also was thinking - when we iterate through the chapterdata objects, are those references to the objects or new copies? If they're references to the objects then we don't gain any memory improvement with this method and we introduce a force unwrap, in which case I'd prefer to go back to just storing the object rather than the index. lmk what you think.

Reply:
Oh it is indeed copied by reference for non primitive data.

Anyways not a big fan of the spread operator with the Array function even in the original

Reply:
```
chapterMap.forEach(mapValue => {
    chapters.push(mapValue.chapter)
})
```

Reply:
ok will revert the index thing since the objects are stored by reference and remove the spread operator in favor of looping through all the elements and pushing